dbt/adapters/firebolt/connections.py

```diff
@@ -162,3 +184,22 @@ def cancel(self, connection: Connection) -> None:
         raise dbt.exceptions.NotImplementedError(
             '`cancel` is not implemented for this adapter!'
         )
+
+
+def _determine_auth(credentials: FireboltCredentials) -> Auth:
+    auth: Auth
```

Comment:
nit: return in every case instead of assigning an returning in the end. This way typing should be fine I believe